CorsixTH/Lua/dialogs/fullscreen/annual_report.lua

```diff
@@ -584,82 +574,64 @@ function UIAnnualReport:drawStatisticsScreen(canvas, x, y)
   -- Helper function to find where the person is in the array.
   -- TODO: This whole sorting thing, it should be possible to do it in a better way?
   local getindex = function(tablename, val)
-    local i = 0
-    local index
     for ind, value in ipairs(tablename) do
-      if value == val then
-        if not index then
-          index = ind
-        end
-        i = i + 1
+      if value[2] == val then
+        return ind
       end
     end
-    return index, i
   end
 
   local row_y = 128
   local row_dy = 15
   local col_x = 190
   local row_no_y = 106
-  local dup_money = 0
-  local dup_salary = 0
-  local dup_cures = 0
-  local dup_deaths = 0
-  local dup_visitors = 0
-  local dup_value = 0
-  for _, player in ipairs(world.hospitals) do
+
+  for idx, player in ipairs(world.hospitals) do
```

Comment:
It'd be nice if this was switched from player to hospital to avoid confusion and be consistent with the loop above. For the same reason `idx` should match `i` above. I prefer `i` or `hosp_index` to `idx`.